### ISSUE TITLE
Stream functionality breaks on Web after `1.9.7`

### ISSUE THREAD

**Issue by iqfareez** [anasfik/openai#57]
## Issue

The chat completion stream on Flutter Web platforms returns all chat at the end. Not streaming.

Using version `dart_openai: ^2.0.1`

See video:

[flutter_chatgpt_notok.webm](https://github.com/anasfik/openai/assets/60868965/f555fdd1-90ed-458e-98e0-99a327b2a763)

Note: Works ok on other platforms (Tested on Windows)

## Expected

After downgrading & pinned the package version: `dart_openai: 1.9.7`. The stream completion on web works normally again.

See video:

[flutter_chatgpt_ok.webm](https://github.com/anasfik/openai/assets/60868965/21605463-6e9a-440f-a1f0-b920e55f64f3)

## Code snippet

```dart
Stream<OpenAIStreamChatCompletionModel> chatStream =
    OpenAI.instance.chat.createStream(
  model: "gpt-3.5-turbo",
  messages: _aiMessages,
);

chatStream.listen((chatStreamEvent) {
  debugPrint(chatStreamEvent.toString());
  // existing id: just update to the same text bubble
  if (chatResponseId == chatStreamEvent.id) {
    chatResponseContent +=
        chatStreamEvent.choices.first.delta.content ?? '';

    _addMessageStream(chatResponseContent);

    if (chatStreamEvent.choices.first.finishReason == "stop") {
      isAiTyping = false;
      _aiMessages.add(OpenAIChatCompletionChoiceMessageModel(
        content: chatResponseContent,
        role: OpenAIChatMessageRole.assistant,
      ));
      chatResponseId = '';
      chatResponseContent = '';
    }
  } else {
    // new id: create new text bubble
    chatResponseId = chatStreamEvent.id;
    chatResponseContent = chatStreamEvent.choices.first.delta.content ?? '';
    onMessageReceived(id: chatResponseId, message: chatResponseContent);
    isAiTyping = true;
  }
});
```

Full code: https://github.com/iqfareez/flutter_chatgpt/blob/main/lib/screens/chat_page.dart

**Comment by elias-avatar** [comment]
Thank you for figuring this out @iqfareez, downgrade to 1.9.7 fixed streaming on web for me

**Comment by khanliKU** [comment]
Did this get fixed? I am using 3.0.0 and Streams don't work on the web.

**Comment by elias-avatar** [comment]
I don't think it was fixed. I've stayed on 1.9.7 for anything web
